Fluid.ViewEngine/FluidViewEngineOptions.cs

```diff
@@ -74,5 +74,15 @@ public class FluidViewEngineOptions
         /// Gets or sets the delegate to execute when a view is rendered.
         /// </summary>
         public RenderingViewDelegate RenderingViewAsync { get; set; }
+
+        /// <summary>
+        /// Represents the method that will generate a cache key for a supplied template path.  Allows templates to be cached based on any external factor.
+        /// </summary>
+        public delegate string TemplateCacheKeyProviderDelegate(string path);
```

Comment:
I'm not sure if it's a good idea to have a delegate, I'm thinking of providing another API for caching or using `IMemoryCache` where the cache key could be provided by the user